aws/lambda/pytorch-auto-revert/pytorch_auto_revert/signal_actions.py

```diff
@@ -378,14 +416,30 @@ def execute_restart(
         ok = True
         if not dry_run:
             try:
-                self._restart.restart_workflow(workflow_target, commit_sha)
+                self._restart.restart_workflow(
+                    workflow_target,
+                    commit_sha,
+                    jobs_to_include=jobs_to_include,
+                    tests_to_include=tests_to_include,
+                )
             except Exception as exc:
                 ok = False
                 notes = str(exc) or repr(exc)
                 logging.exception(
                     "[v2][action] restart for sha %s: exception while dispatching",
                     commit_sha[:8],
                 )
+
+        # Build notes with filter information
+        notes_parts = []
+        if jobs_to_include:
+            notes_parts.append(f"jobs_filter={','.join(jobs_to_include)}")
+        if tests_to_include:
+            notes_parts.append(f"tests_filter={','.join(tests_to_include)}")
+        if notes:  # Error message from exception
+            notes_parts.append(notes)
+        notes = "; ".join(notes_parts) if notes_parts else ""
```

Comment:
you are overwriting notes here, it can have been set in other parts of the code, maybe you want to create notes_parts right at the function start and poppulate it?